Added new quickstart security quick starts< ejb-security, servlet-securi...

agabios opened the pull request:
Quick starts 15 and 16

Commit:
Added new quickstart security quick starts< ejb-security, servlet-security>

sgilda commented:
Hi Sherif,
I tested the ejb-security quickstart and it works great. I do have a couple of comments:

-- First, I noticed you have 2 quickstarts in this pull. In the future, it is probably better to create a separate branch for each quickstart.  It's cleaner and that way we can push one in when it's been approved and ready. When you have multiple quickstarts in on pull like this, it can take longer to get them both in.

-- The README.md file title should be a header and follow the format of "quickstart-name: Description of quickstart". For example:     ejb-security:  Using JEE Declarative Security to Control Access to EJB 3 

The README file needs a little more details so it isn't too confusing for the users. It doesn't display the XML snippets correctly. I found this web site useful to test the markdown syntax: 

```
http://daringfireball.net/projects/markdown/dingus
```

I will send a revised README.md for the ejb-security example by email to show you what I mean.

I will start testing the servlet-security quickstart next.

Pete, who should do the code review for these quickstarts? Anil Saldhana?

sgilda commented:
Hi Sherif,
I tested the servlet-security quickstart and it works great too.
Please modify the README for that one too. Specifically:
-- Make the title conform to the standard described here: https://docs.jboss.org/author/display/AS71/Writing+a+quickstart
-- Reorganize the instructions so it flows better for the user
-- Give more specific details where to paste the XML and paths to where the files are located
-- More details on the output when testing for a security failure
-- Fix the markdown syntax so it renders correctly in HTML. Make sure the XML snippets appear. You can test it out at this URL: http://daringfireball.net/projects/markdown/dingus

Thanks!

Pete and Anil will need to do a code review.

sgilda commented:
Hi Anil,
Can you do a code review for this quickstart?
Thanks, Sande

On 03/01/2012 08:57 PM, smakary wrote:

> Quick starts 15 and 16
> 
> You can merge this Pull Request by running:
> 
>    git pull https://github.com/smakary/quickstart security
> 
> Or you can view, comment on it, or merge it online at:
> 
>    https://github.com/jbossas/quickstart/pull/131
> 
> -- Commit Summary --
> - Added new quickstart security quick starts<  ejb-security, servlet-security>
> 
> -- File Changes --
> 
> A ejb-security/README.md (61)
> A ejb-security/pom.xml (114)
> A ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/CallSecuredEJBServlet.java (91)
> A ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/SecuredEJB.java (40)
> A ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/SecuredEJBBean.java (67)
> A ejb-security/src/main/webapp/WEB-INF/beans.xml (7)
> A ejb-security/src/main/webapp/WEB-INF/classes/roles.properties (1)
> A ejb-security/src/main/webapp/WEB-INF/classes/users.properties (1)
> A ejb-security/src/main/webapp/WEB-INF/jboss-web.xml (6)
> A ejb-security/src/main/webapp/WEB-INF/web.xml (24)
> A ejb-security/src/main/webapp/index.html (7)
> A servlet-security/README.md (64)
> A servlet-security/pom.xml (110)
> A servlet-security/src/main/java/org/jboss/as/quickstarts/servlet_security/SecuredServlet.java (85)
> A servlet-security/src/main/webapp/WEB-INF/beans.xml (7)
> A servlet-security/src/main/webapp/WEB-INF/classes/roles.properties (1)
> A servlet-security/src/main/webapp/WEB-INF/classes/users.properties (1)
> A servlet-security/src/main/webapp/WEB-INF/jboss-web.xml (6)
> A servlet-security/src/main/webapp/WEB-INF/web.xml (24)
> A servlet-security/src/main/webapp/index.html (7)
> 
> -- Patch Links --
> 
>    https://github.com/jbossas/quickstart/pull/131.patch
>    https://github.com/jbossas/quickstart/pull/131.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/131

sgilda commented:
Sherif, your code is very nicely commented and easy to follow. :-)

sgilda commented:
Hi Sherif, 
You need to use the AS 7 versions in your pom.xml files. When we merge your quickstarts to EAP 6, we modify the versions at that time to comply with EAP 6. 

You can find most of the version to use on this page: 
   https://docs.jboss.org/author/display/AS71/Maven+POM+Versions+Checklist

Please make the following changes to both of your pom.xml files:
org.jboss.as.quickstarts groupId: change the version from 7.0.2.CR4 to 7.1.1-SNAPSHOT
jboss-javaee-6.0 artifiactId: change the version from 3.0.0.Beta1-redhat-1 to 3.0.0.Beta1
jboss-as-maven-plugin (artifactId) : change the version from 7.1.0.CR1 to 7.1.0.Final 

Thanks, Sande

Commit:
Fixed pom and README.md for ejb-security

Commit:
Fixed the readme and the pom files for ejb-security

Commit:
Fixed readme and pom for ejb-security

Commit:
Fixed readme and pom for servlet-security

agabios commented:
Hi Sande,
-I will make sure to create a dedicated branch for the future quick start
-I fixed the pom and the readme for the ejb-security and committed changes
-I fixed the pom and the readme for the servlet-security and committed changes
Thanks for your help.
Sherif

sgilda commented:
Hi Sherif, I will need to check with Pete on the mistaken commit  8d96001 of the root README.md and pom.xml files. I'm not sure how to handle that.
Thanks, Sande

agabios commented:
Hi Sande,
Even if in my own local branch?
Sorry about that.
Thanks,

------Original Message------
From: sgilda
To: Sherif Makary
Subject: Re: [quickstart] Added new quickstart security quick starts< ejb-security, servlet-securi... (#131)
Sent: Mar 4, 2012 6:30 PM

Hi Sherif, I will need to check with Pete on the mistaken commit  8d96001 of the root README.md and pom.xml files. I'm not sure how to handle that.
Thanks, Sande

---

Reply to this email directly or view it on GitHub:
https://github.com/jbossas/quickstart/pull/131#issuecomment-4314624

Sent from my BlackBerry device on the Rogers Wireless Network

sgilda commented:
Yes. It's included in the pull (see above in the list of the 4 commits) and I'm not sure how to avoid merging it too.

sgilda commented:
Sherif, to be clearer, when we process this request, it will overwrite the root README.md and pom.xml files with these bad versions. I'm not skilled enough in git to figure out how to get rid of that incorrect commit. :-)

agabios commented:
Thanks Sande,
I'm  doing some research on selective pull, I guess worst case to just overwrite this version with the correct ones, and we can just wait for Pete's input.
Thanks,
Sherif

Commit:
Fixed comments

Commit:
Fixed security getting started reference

Commit:
Fixed deployment target

Commit:
Fixed deployment url

sgilda commented:
Looking at it now.

sgilda commented:
Hi Sherif,
I am able to test and deploy both quickstarts, but I am not able to run them.

First, I know I helped rewrite your README file, but now I'm finding it confusing. It's unclear which steps need to be done by the user and which were done in the code provided in the quickstart.

For example, maybe the README should be changed to read:

To implement EJB security:
1. The application must add a security-domain in the jboss-web.xml file. Refer to the /webapp/WEB-INF/jboss-web.xml -war packaging for the security domain xml. Alternatively the application can use jboss-ejb.xml for jar packaging,  
2.  The application must configure users.properties and roles.properties files in WEB-INF/classes directory of the web application
3. The administrator must configure a security domain in the JBoss AS standalone.xml file.

To implement EJB declarative security, the application must:
1. Add security annotations to your EJB declaration
2. Make sure the allowed user role is the same as the role defined in roles.properties file
3. Make sure the security domain referenced in the jboss-web.xml is defined in the JBoss AS standalone.xml configuration file.  <followed by the instructions to add the snippet>

I added the XML snippet to my configuration file. My attempts to log in with admin/admin gave me:
        HTTP Status 404 - /jboss-as-ejb-security/EJB-Security
        type Status report
        message /jboss-as-ejb-security/EJB-Security
        description The requested resource (/jboss-as-ejb-security/EJB-Security) is not available.
I don't see anything at all in the server log.

Do I need to enable the admin user for this to work? If so, I think we need specific instructions. I tried adding an admin user with the bin/add-user.sh script. It wouldn't allow me to add admin/admin, so I added admin/password. The quickstart still didn't work.

Am I missing a step?

Thanks, Sande

sgilda commented:
Another oddity, I just deployed servlet-security and when I accessed the application using this URL: http://localhost:8080/jboss-as-servlet-security/SecuredServlet, I was NOT prompted for a login and it went directly to this:

Successfully called Secured Servlet
Principal : admin
Remote User : admin
Authentication Type : BASIC

sgilda commented:
I think I figured out one of the problems. This is what I am doing:
--  I hit the URL: http://localhost:8080/jboss-as-ejb-security
--  I get the login prompt.
--  I login as admin/admin
--  I get the HTTP Status 404.
--  I noticed the URL was redirected to http://localhost:8080/jboss-as-ejb-security/EJB-Security
--  In the quickstart README, the instructions say to access URL: http://localhost:8080/jboss-as-ejb-security/CallSecuredEJBServlet
-- When I access that URL, it goes directly to the "Successfully called Secured EJB" page. I assume its because I logged in at the previous prompt.

Should the URL in the index.html file be CallSecuredEJBServlet instead of EJB-Security?
Then the instructions could be consistent with the other quickstarts and the README could instruct you to access the application at http://localhost:8080/jboss-as-ejb-security.

I see a similar situation with the URL in the servlet-security index.html file.

Commit:
Fixed readme and index.html for ejb-security

Commit:
Fixed readme and index.html for servlet-security

Commit:
Update servlet-security/README.md

agabios commented:
Hi Sande,
-You got the browser login cache right, it's also in the readme, you need to clear the active login cache
-I fixed the index.html for both ejb security and servlet security
-I updated the readme files to be more clear in terms of which actions needs to be taken by the quick start users
Please have a look and let me know.
Thanks,
Sherif

pmuir commented:
Sherif, could you squash all your commit using git rebase?

pmuir commented:
Regarding the problem with the root readme, the best bet is just to drop all changes and then make what is needed again.

Once you have squashed all commits, just rerun the rebase, and edit the commit, removing the root README from the changeset.

agabios commented:
Hi Pete,
Would that be $git rebase origin
Thanks,
Sherif

pmuir commented:
You need to use interactive rebase. Here is a reasonable description - http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

pmuir commented on ejb-security/README.md:
```diff
@@ -0,0 +1,81 @@
+ejb-security:  Using JEE Declarative Security to Control Access to EJB 3
+====================
+Author: Sherif Makary, RH MW SA
```
I've no problem with you including your "provenance" but I think that you need to avoid acronyms - the average joe on the street won't know what these letters mean ;-) Spell it out - Red Hat Middleware Solutions Architect

agabios replied:
Sorry, I was aiming for EAP6, I'll fix those.

pmuir commented on ejb-security/README.md:
```diff
@@ -0,0 +1,81 @@
+
+To implement EJB security, you need to:
+
+1. The application will use a security domain that needs to defined in the application server standalone.xml as follows,
```
This doesn't flow "To implement EJB security, you need to The application will use a security domain that needs to defined in the.... " If you start a list with the word "to" you need to start each item with a verb.

agabios replied:
Typo on my part, I'll fix it

pmuir replied:
BTW this applies to each item in the list.

agabios replied:
If I change this to "The Steps required to implement xyz" would that work, also, note, that the user will have to action one step only which is adding the security domain to the standalone.xml, other than that, all implemented in the example code, thoughts.

pmuir commented on ejb-security/README.md:
```diff
@@ -0,0 +1,81 @@
+                 </authentication>
+             </security-domain>
+
+2. A security-domain reference is added to /webapp/WEB-INF/jboss-web.xml -war packaging-, you can use jboss-ejb.xml for jar packaging  
```
I think you need to make it clearer that using jboss-ejb.xml is an alternative used in when you package using an EAR or a standalone EJB jar. I'm not sure you should even mention jboss-ejb.xml as it's outside the scope of this quickstart (which uses wars).

agabios replied:
The war EJB packaging is new to the EAP 5 base, thought to mention just in case they would like to repackage for jar/ear

pmuir replied:
Ok, but it needs to be a lot clearer that it's an alternative.

agabios replied:
Fixed

pmuir commented on ejb-security/README.md:
```diff
@@ -0,0 +1,81 @@
+             </security-domain>
+
+2. A security-domain reference is added to /webapp/WEB-INF/jboss-web.xml -war packaging-, you can use jboss-ejb.xml for jar packaging  
+3. The security domain will use users.properties and roles.properties files to verify users and roles, the files added to WEB-INF/classes directory of your web application
```
"...the >>>are<<< added to..."

agabios replied:
Fixed

pmuir commented on ejb-security/README.md:
```diff
@@ -0,0 +1,81 @@
+
+2. A security-domain reference is added to /webapp/WEB-INF/jboss-web.xml -war packaging-, you can use jboss-ejb.xml for jar packaging  
+3. The security domain will use users.properties and roles.properties files to verify users and roles, the files added to WEB-INF/classes directory of your web application
+4. Security annotations added to the EJB declaration
```
"... >>>are<<< added to..."

agabios replied:
Fixed

pmuir commented on ejb-security/README.md:
```diff
@@ -0,0 +1,81 @@
+2. A security-domain reference is added to /webapp/WEB-INF/jboss-web.xml -war packaging-, you can use jboss-ejb.xml for jar packaging  
+3. The security domain will use users.properties and roles.properties files to verify users and roles, the files added to WEB-INF/classes directory of your web application
+4. Security annotations added to the EJB declaration
+5. Please note the allowed user role in the annotation -`@RolesAllowed`- is the same as the role defined in roles.properties file
```
If this is a note, I don't think it belongs in the list.

agabios replied:
Fixed

pmuir commented on ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/CallSecuredEJBServlet.java:
```diff
@@ -0,0 +1,90 @@
+ */
+@SuppressWarnings("serial")
+@WebServlet("/CallSecuredEJBServlet")
+
```
Superfluous new line

agabios replied:
Fixed

pmuir commented on ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/CallSecuredEJBServlet.java:
```diff
@@ -0,0 +1,90 @@
+        //Get user name from login principal
+    	remoteUser = req.getRemoteUser();
+    	//Get authentication type
+    	authType = req.getAuthType();
```
Something has gone wrong with the indent here.

agabios replied:
Fixed

pmuir commented on ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/CallSecuredEJBServlet.java:
```diff
@@ -0,0 +1,90 @@
+   @Override
+   protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+      PrintWriter writer = resp.getWriter();
+      //Get user principal
```
This comment doesn't make sense to me - you just set three strings to null straight after?

agabios replied:
Fixed

pmuir commented on ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/SecuredEJBBean.java:
```diff
@@ -0,0 +1,67 @@
+ * 
+ */
+
+public @Stateless class SecuredEJBBean implements SecuredEJB {
```
Annotations shouldn't be mixed in with keywords.

agabios replied:
Fixed

pmuir commented on ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/SecuredEJB.java:
```diff
@@ -0,0 +1,40 @@
+public interface SecuredEJB {
+	public String getSecurityInfo();
+
+}
```
Wonder whether using a no-interface-view EJB would simplify things here?

pmuir commented on ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/SecuredEJBBean.java:
```diff
@@ -0,0 +1,67 @@
+
+
+
+}
```
Indenting and whitespace is cocked up in this file

agabios replied:
Fixed

pmuir commented on ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/SecuredEJBBean.java:
```diff
@@ -0,0 +1,67 @@
+		//Session context injected using the resource annotation
+		principal = ctx.getCallerPrincipal();
+		
+   		return principal.toString();
```
This is a very odd use of variables. Why not just

```
return ctx.getCallerPrincipal().toString();
```

and remove the principal field?

agabios replied:
I thought it's good for illustration, especially with the  session context comment

pmuir replied:
Not really. The session context is indeed injected, but you never use the principal field as a field. Furthermore, the bean is a stateless bean, so it's very bad practice to assign values to fields.

I've no problem if you want to make the Principal a method local variable, to enable commenting it better (you would need more comments than you have now thought to justify this).

sgilda replied:
I thought this pull was closed and replaced by 
https://github.com/jbossas/quickstart/pull/147 and 
https://github.com/jbossas/quickstart/pull/148, but I don't see a record 
of it.

Sherif, didn't you create separate branches for each of the quickstarts 
and submit new pull requests?

On 03/19/2012 06:18 AM, Pete Muir wrote:

> > -   
> > -    /**
> > -     *<p>
> > -     \* sample Secured ejb method using security annotations
> > -     *</p>
> > -     *
> > -     */ 
> > -   
> > -    @Override
> > -    @RolesAllowed({"gooduser"})
> > -    public String getSecurityInfo()
> > -       {
> > -        //Session context injected using the resource annotation
> > -        principal = ctx.getCallerPrincipal();
> > -   
> > -        return principal.toString();
> >   Not really. The session context is indeed injected, but you never use the principal field as a field. Furthermore, the bean is a stateless bean, so it's very bad practice to assign values to fields.
> 
> I've no problem if you want to make the Principal a method local variable, to enable commenting it better (you would need more comments than you have now thought to justify this).
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/131/files#r573353

agabios replied:
Yes, I did, and I sent few emails about this, and I haven't changed anything since then, not sure why you don't see the new branches.

sgilda replied:
I do see them.
Pete, I think you're looking at the old code base, although the new code 
base may contain the same code. :-)

On 03/19/2012 07:49 AM, Sherif Makary wrote:

> > -   
> > -    /**
> > -     *<p>
> > -     \* sample Secured ejb method using security annotations
> > -     *</p>
> > -     *
> > -     */ 
> > -   
> > -    @Override
> > -    @RolesAllowed({"gooduser"})
> > -    public String getSecurityInfo()
> > -       {
> > -        //Session context injected using the resource annotation
> > -        principal = ctx.getCallerPrincipal();
> > -   
> > -        return principal.toString();
> >   Yes, I did, and I sent few emails about this, and I haven't changed anything since then, not sure why you don't see the new branches.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/131/files#r573557

agabios replied:
I did a lot of fixes, except the ones that Pete highlighted, please switch to the new branches, it's cleaner this way and it won't have the root pom/readme issues.

sgilda replied:
Can we stop adding comments to this closed pull? It's getting confusing. :-)
Just add the comments to the appropriate new pull and reference the 
comment in the closed pull.

On 03/19/2012 08:13 AM, Sherif Makary wrote:

> > -   
> > -    /**
> > -     *<p>
> > -     \* sample Secured ejb method using security annotations
> > -     *</p>
> > -     *
> > -     */ 
> > -   
> > -    @Override
> > -    @RolesAllowed({"gooduser"})
> > -    public String getSecurityInfo()
> > -       {
> > -        //Session context injected using the resource annotation
> > -        principal = ctx.getCallerPrincipal();
> > -   
> > -        return principal.toString();
> >   Regarding the "Principal", the idea was to show how to get the principal to use it in is in role, I'll change the variable scope to local variable in the new branch, also I'm going to remove the CDI support.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/131/files#r573591

pmuir commented on ejb-security/pom.xml:
```diff
@@ -0,0 +1,114 @@
+                  up! -->
+               <failOnMissingWebXml>false</failOnMissingWebXml>
+            </configuration>
+         </plugin>
```
You should remove the war plugin configuration, as you _do_ have a `web.xml`

agabios replied:
Fixed

pmuir commented on ejb-security/src/main/webapp/WEB-INF/web.xml:
```diff
@@ -0,0 +1,24 @@
+	  <login-config>
+    	<auth-method>BASIC</auth-method>
+    	<realm-name>WebSecurityBasic</realm-name>
+  	  </login-config>
```
indent problem.

agabios replied:
Fixed

pmuir commented on ejb-security/src/main/java/org/jboss/as/quickstarts/ejb_security/CallSecuredEJBServlet.java:
```diff
@@ -0,0 +1,90 @@
+
+   static String PAGE_FOOTER = "</body></html>";
+   
+   //Injecting the Secured EJB
```
I think this should say "Inject the secured EJB"

agabios replied:
Fixed

pmuir commented on ejb-security/README.md:
```diff
@@ -0,0 +1,81 @@
+3. The security domain will use users.properties and roles.properties files to verify users and roles, the files added to WEB-INF/classes directory of your web application
+4. Security annotations added to the EJB declaration
+5. Please note the allowed user role in the annotation -`@RolesAllowed`- is the same as the role defined in roles.properties file
+
```
Shouldn't you make a note about the `web.xml` here as well?

agabios replied:
Sorry Pete, can you please clarify?

pmuir replied:
You put some stuff in web.xml for this example. I think you should make a note of that in this list.

pmuir commented on ejb-security/src/main/webapp/WEB-INF/web.xml:
```diff
@@ -0,0 +1,24 @@
+    xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd"
+    version="3.0">
+
+  <security-constraint>
```
I think this could do with some comments

pmuir commented on pom.xml:
```diff
@@ -1,83 +1,110 @@
+               <target>1.6</target>
+            </configuration>
+         </plugin>
+      </plugins>
```
You need to revert all changes here and start again.

agabios replied:
Pete,
Can you please clarify this one?

pmuir replied:
You've made a lot of changes to the root POM.xml that are uncessary. You simply need to add your new quickstart. Revert changes to the file.

pmuir commented on servlet-security/README.md:
```diff
@@ -0,0 +1,83 @@
+
+To implement Servlet security, you need to:
+
+1. The application will use a security domain that needs to defined in the application server standalone.xml as follows,
```
As with ejb-security, you need to use a verb to start each item in a list that starts "to"

agabios replied:
Fixed

pmuir commented on servlet-security/README.md:
```diff
@@ -0,0 +1,83 @@
+             </security-domain>
+
+2. A security-domain reference is added to /webapp/WEB-INF/jboss-web.xml 
+3. A security-constraints is added to the /webapp/WEB-INF/web.xml
```
If something is plural, use The, not A

pmuir commented on servlet-security/pom.xml:
```diff
@@ -0,0 +1,110 @@
+               <!-- Java EE 6 doesn't require web.xml, Maven needs to catch 
+                  up! -->
+               <failOnMissingWebXml>false</failOnMissingWebXml>
+            </configuration>
```
You do use a web.xml so this plugin config can be removed...

agabios replied:
Fixed

pmuir commented on servlet-security/src/main/java/org/jboss/as/quickstarts/servlet_security/SecuredServlet.java:
```diff
@@ -0,0 +1,85 @@
+@SuppressWarnings("serial")
+@WebServlet("/SecuredServlet")
+@ServletSecurity(@HttpConstraint(rolesAllowed = { "gooduser" }))
+
```
Superfluous new line

pmuir commented on servlet-security/src/main/java/org/jboss/as/quickstarts/servlet_security/SecuredServlet.java:
```diff
@@ -0,0 +1,85 @@
+   @Override
+   protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+      PrintWriter writer = resp.getWriter();
+      //Get user principal
```
As before, this comment doesn't make much sense

agabios replied:
Fixed

pmuir commented on servlet-security/src/main/java/org/jboss/as/quickstarts/servlet_security/SecuredServlet.java:
```diff
@@ -0,0 +1,85 @@
+      
+      	//Get security principal  
+      	principal = req.getUserPrincipal();
+        //Get user name from login principal
```
indent issue

agabios replied:
Fixed

pmuir commented on servlet-security/src/main/webapp/WEB-INF/beans.xml:
```diff
@@ -0,0 +1,7 @@
+      http://java.sun.com/xml/ns/javaee 
+      http://java.sun.com/xml/ns/javaee/beans_1_0.xsd">
+</beans>
\ No newline at end of file
```
I don't see why you are activating CDI, you don't use it...

pmuir commented on ejb-security/src/main/webapp/WEB-INF/beans.xml:
```diff
@@ -0,0 +1,7 @@
+   xsi:schemaLocation="
+      http://java.sun.com/xml/ns/javaee 
+      http://java.sun.com/xml/ns/javaee/beans_1_0.xsd">
+</beans>
```
I don't see why you are activating CDI, you aren't using it.

agabios replied:
I thought to we enable this in general for all quick starts in case the user want to extend to CDI

pmuir replied:
A quickstart should show the minimum code and services required. If you aren't using CDI (you aren't), then don't enable CDI.

pmuir commented:
Sherif, once you've addressed my comments, can you comment back and I'll re-review. I'm also searching for a SME for the security aspects.

darranl commented:
I will have a look shortly at the security aspects, need to do some testing in this area anyway so an ideal time to use the quickstart.

darranl commented:
I haven't started to review the full details yet but one quick question - is there any reason now to avoid the default security domain now included in AS7 by default?

agabios commented:
-When I started with AS 7.0, I wasn't sure which one is the default, and the application-user.properties didn't work for me with the encrypted password
-Also, I thought it's a good opportunity to educate the users about the new login config and how to add a security domain.
Thanks,
Sherif

darranl commented:
The problem here is that the approach you are using you are encouraging them to use properties files with passwords stored entirely on the clear - for the default domain we now encourage the use of pre digested (not encrypted) passwords and also provide a command line utility to add the users and their roles to those files.

agabios commented:
This is just for illustration purposes, usually people use other means in production, such as LDAP, DB or cert, this is a common theme in other security quick starts, I can add the disclaimer if you wish.

sgilda commented:
Darran,

I am in the process of modifying the README at the root of the 
quickstarts to tell them how to add a Management user and how to add an 
Application user using the bin/add-user.sh script. I don't think we 
should be telling them to modify properties files.

What should we use for the Management user? Is there a preferred 
userid/password for this?

Thanks,
Sande

On 03/14/2012 10:47 AM, Darran Lofthouse wrote:

> The problem here is that the approach you are using you are encouraging them to use properties files with passwords stored entirely on the clear - for the default domain we now encourage the use of pre digested (not encrypted) passwords and also provide a command line utility to add the users and their roles to those files.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/131#issuecomment-4499466

agabios commented:
This is would be an app user and it wouldn't use the management, last time I used the app user, the Basic browser authentication didn't work with the encrypted password, not sure if that was tested and fixed.

darranl commented:
Do you have the config that you tested with it?  It is more likely that the config was wrong rather than there being something that needed a fix as this is already being widely used.

agabios commented:
I'll send the standalone.xml, I was getting default user properties not found with IO exception and when I move the properties files to class path, I couldn't login till I replaced the encrypted password with clear.
I can test again later in the evening.
My early test was 8 weeks ago before I venture with custom security domain.

darranl commented:
Yeah the standalone.xml would be useful to double check - from your description it does sound like a config issue.

sgilda commented:
Closing this pull request. It has been replaced by 2 separate pull requests:
-- https://github.com/jbossas/quickstart/pull/147
-- https://github.com/jbossas/quickstart/pull/148
